http/codegen/service_data.go

```diff
@@ -1639,21 +1701,32 @@ func extractQueryParams(a *design.MappedAttributeExpr, serviceType *design.Attri
 	return params
 }
 
-func extractHeaders(a *design.MappedAttributeExpr, serviceType *design.AttributeExpr, req bool, scope *codegen.NameScope) []*HeaderData {
+func extractHeaders(a *design.MappedAttributeExpr, serviceType *design.AttributeExpr, req, viewed bool, scope *codegen.NameScope) []*HeaderData {
 	var headers []*HeaderData
 	codegen.WalkMappedAttr(a, func(name, elem string, required bool, c *design.AttributeExpr) error {
 		var (
 			varn    = codegen.Goify(name, false)
 			arr     = design.AsArray(c.Type)
 			typeRef = scope.GoTypeRef(c)
+
+			vcode string
 		)
-		if a.IsPrimitivePointer(name, true) {
+		if a.IsPrimitivePointer(name, true) || viewed {
 			typeRef = "*" + typeRef
 		}
 		fieldName := codegen.Goify(name, true)
 		if !design.IsObject(serviceType.Type) {
 			fieldName = "" // result is initialized directly from header
 		}
+		if viewed {
+			// In a viewed result type the attributes are always pointers.
+			required = false
+			// No need to generate validation code since the validation is performed
+			// by the viewed result type.
+			vcode = ""
```

Comment:
Are validations other than required run somewhere else?

Reply:
Yup.. the validation function in the views package runs all validations.